main.js

```diff
@@ -65,10 +60,14 @@ let baseItemWidth = 300;
 let baseItemHeight = 250;
 let baseGridSize = 800;
 
+var currentWidgets = [];
```

Comment:
Could this be `const`? I only see that we are calling `push` on it, but never re-assigning it.

Same for `gridItemDict` on the line below.

Reply:
👍